src/planwise/client/components/nav.cljs

```diff
@@ -3,12 +3,17 @@
 
 ;; Navigation components
 
-(defn- li-menu-item [{:keys [selected item href title icon]}]
+(defn- li-menu-item [{:keys [selected item href title icon tab-number wizard-state]}]
   (let [is-selected? (or (= selected item)
                          (item selected))]
-    [:li {:class (when is-selected? "active")}
+    [:li {:class (str
+                  (when is-selected? "active ")
+                  (when wizard-state (str (name wizard-state) " wizard")))}
      [:a {:href href}
       (some-> icon (components/icon "icon-small"))
+      (when (and wizard-state (> tab-number 1)) [:div.separator])
+      (when (and wizard-state (= wizard-state :visited)) (components/icon :check-circle-wizard "icon-small wizard-check"))
+      (when (and wizard-state (not= wizard-state :visited)) [:div.wizard-number tab-number])
```

Comment:
I don't like coupling the `li-menu-item` function with the wizard. I'd rather have the function accept a generic content to be rendered along with or instead of the title, and work the wizard-state from outside the function. You could even have a `li-wizard-menu-item` function that calls `li-menu-item` with this.